src/processing/queryprocessor.cpp

```diff
@@ -59,8 +59,8 @@ std::unique_ptr<QueryProcessor::QueryResult> QueryProcessor::QueryResult::polygo
 std::unique_ptr<QueryProcessor::QueryResult> QueryProcessor::QueryResult::plot(const std::string &plot, const QueryRectangle &qrect, std::unique_ptr<ProvenanceCollection> provenance) {
 	return std::unique_ptr<QueryResult>(new QueryResult(Query::ResultType::PLOT, nullptr, std::move(provenance), plot, "", qrect));
 }
-std::unique_ptr<QueryProcessor::QueryResult> QueryProcessor::QueryResult::error(const std::string &error, const QueryRectangle &qrect) {
-	return std::unique_ptr<QueryResult>(new QueryResult(Query::ResultType::ERROR, nullptr, nullptr, "", error, qrect));
+std::unique_ptr<QueryProcessor::QueryResult> QueryProcessor::QueryResult::error(const std::string &error, const QueryRectangle &qrect, MappingExceptionType errorType) {
```

Comment:
Why pass message and type separately instead of as MappingException?

Reply:
The method is also used when a `std::exception` is caught, where the ExceptionType is set to confidential by hand. And because it's only called twice, an overload seems unnecessary to me.

Reply:
I think it would be fine to accept only MappingException. std::exception could be wrapped

Reply:
Okay, the change is pushed.